### PR TITLE
Add dm e2e test

### PR DESCRIPTION
**Pull request by GMHDBJD**
- add e2e test for with dm jobmaster

### CODE REVIEW

File: jobmaster/dm/checkpoint/agent.go [hunk 1]
```diff
@@ -178,9 +188,10 @@ func onlineDDLName(cfg *config.JobCfg) string {
 }
 
 func isLoadFresh(ctx context.Context, taskCfg *config.TaskCfg, db *conn.BaseDB) (bool, error) {
-	query := "SELECT status FROM %s WHERE `task_name` = ? AND `source_name` = ?"
+	// nolint:gosec
+	query := fmt.Sprintf("SELECT status FROM %s WHERE `task_name` = ? AND `source_name` = ?", loadTableName((*config.JobCfg)(taskCfg)))
```
Comment (GMHDBJD) [inline jobmaster/dm/checkpoint/agent.go:192]: Both %s and ? tasks no effect for tablename, so use fmt instead.

**Comment by GMHDBJD** [comment]
cc @lance6716

File: dm/unit.go [hunk 2]
```diff
@@ -77,19 +87,42 @@ func (u *unitHolder) getResult() (bool, *pb.ProcessResult) {
...
+		statusBytes, err := json.Marshal(status)
+		if err != nil {
+			return err
+		}
 		s := libModel.WorkerStatus{
-			Code: libModel.WorkerStatusNormal,
+			Code:     libModel.WorkerStatusNormal,
+			ExtBytes: statusBytes,
 		}
 		// nolint:errcheck
```
Comment (lance6716) [inline dm/unit.go:111]:
```suggestion
```

File: dm/unit.go [hunk 3]
```diff
@@ -116,24 +151,44 @@ func (u *unitHolder) tryUpdateStatus(ctx context.Context, base lib.BaseWorker) e
...
 			Code:         libModel.WorkerStatusError,
 			ErrorMessage: unit.JoinProcessErrors(result.Errors),
+			ExtBytes:     statusBytes,
 		}
 		// TODO: UpdateStatus too frequently?
 		// nolint:errcheck
 		_ = base.UpdateStatus(ctx, s)
 		return nil
 	case worker.ResumeDispatch:
+		u.unit.Pause()
```
Comment (lance6716) [inline dm/unit.go:176]: maybe we can check if the unit has been paused by ResumeSkip. But it should also be OK that we Pause twice

File: jobmaster/dm/checkpoint/agent.go [hunk 4]
```diff
@@ -130,13 +134,19 @@ func (c *AgentImpl) IsFresh(ctx context.Context, workerType lib.WorkerType, task
 	return isSyncFresh(ctx, task.Cfg, db)
 }
 
+func createMetaDatabase(ctx context.Context, cfg *config.JobCfg, db *conn.BaseDB) error {
+	query := fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s", cfg.MetaSchema)
```
Comment (lance6716) [inline jobmaster/dm/checkpoint/agent.go:138]: should backquote it. we can use tiflow/pkg/quotes.QuoteName or dbutil like below

File: jobmaster/dm/message_agent.go [hunk 5]
```diff
@@ -71,7 +71,9 @@ func (agent *MessageAgent) CreateWorker(ctx context.Context, taskID string, work
 	if _, ok := agent.sendHandles.Load(taskID); ok {
 		return "", errors.Errorf("worker for task %s already exist", taskID)
 	}
-	return agent.master.CreateWorker(workerType, taskCfg, 1)
+	// TODO: deprecated subtask cfg
```
Comment (lance6716) [inline jobmaster/dm/message_agent.go:74]: what's the plan for this?

**Review by lance6716 — APPROVED** [review]

File: jobmaster/dm/message_agent.go (hunk 5 above)
Comment (GMHDBJD) [reply jobmaster/dm/message_agent.go:74]: May not

**Comment by GMHDBJD** [comment]
cc @amyangfei @liuzix

**Review by amyangfei — APPROVED** [review]

**Pull request merged**